### PR TITLE
feat(Ruler):add new component

### PR DESCRIPTION
**Pull request by supergaojian**
add new component ruler

### CODE REVIEW

File: components/ruler/index.vue [hunk 1]
```diff
@@ -0,0 +1,357 @@
+<template>
+  <div
+    class="md-ruler"
+    @touchstart="$_startDrag"
+    @touchend="$_stopDrag"
```
Comment (moyus) [inline components/ruler/index.vue]: miss `touchcancel`

File: components/ruler/index.vue [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+    },
+    max: {
+      type: Number,
+      default: 100,
+    },
+  },
+
+  data() {
+    return {
+      ctx: null, // canvas
```
Comment (moyus) [inline components/ruler/index.vue]: data is only for ui reactive, remove `ctx` here and assign `ctx` under `this` context directly.

File: components/ruler/index.vue [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+      }
+
+      event.preventDefault()
+      event.stopPropagation()
+      this.scroller.doTouchStart(event.touches, event.timeStamp)
+
+      this.isDragging = true
+      this.isScrolling = true
+
+      window.addEventListener('mousemove', this.$_onDrag)
```
Comment (moyus) [inline components/ruler/index.vue]: If this component is only for mobile, there is no need `mousemove` event. If you insist, attach `mousedown` event listener as well.

File: components/ruler/index.vue [hunk 4]
```diff
@@ -0,0 +1,357 @@
...
+    },
+
+    $_stopDrag(event) {
+      event.preventDefault()
+      event.stopPropagation()
+      this.isDragging = false
+
+      this.scroller.doTouchEnd(event.timeStamp)
+
+      window.removeEventListener('mousemove', this.$_onDrag)
```
Comment (moyus) [inline components/ruler/index.vue]: If this component is only for mobile, there is no need `mousemove` event. If you insist, attach `mouseup` event listener as well.

File: components/ruler/index.vue [hunk 5]
```diff
@@ -0,0 +1,357 @@
...
+      this.scroller.doTouchStart(event.touches, event.timeStamp)
+
+      this.isDragging = true
+      this.isScrolling = true
+
+      window.addEventListener('mousemove', this.$_onDrag)
+      window.addEventListener('touchmove', this.$_onDrag)
+    },
+
+    $_onDrag(event) {
```
Comment (moyus) [inline components/ruler/index.vue]: Consider `throttle`?

File: components/ruler/index.vue (hunk 2 above)
Comment (supergaojian) [reply components/ruler/index.vue]: I don't think so data is only for ui reactive.Put `ctx` here for other developer understand where the `ctx` is defined.

File: components/ruler/index.vue (hunk 3 above)
Comment (supergaojian) [reply components/ruler/index.vue]: Maybe you are right.The component is only for mobile, so remove the mouse event.

**Pull request merged**